crates/storage/api.rs

```diff
@@ -1,361 +1,142 @@
 
-    /// Sets the last key fetched from the state trie being fetched during snap sync
-    async fn set_state_trie_key_checkpoint(
-        &self,
-        last_keys: [H256; STATE_TRIE_SEGMENTS],
-    ) -> Result<(), StoreError>;
-
-    /// Gets the last key fetched from the state trie being fetched during snap sync
-    async fn get_state_trie_key_checkpoint(
-        &self,
-    ) -> Result<Option<[H256; STATE_TRIE_SEGMENTS]>, StoreError>;
-
-    /// Sets the state trie paths in need of healing
-    async fn set_state_heal_paths(&self, paths: Vec<(Nibbles, H256)>) -> Result<(), StoreError>;
-
-    /// Gets the state trie paths in need of healing
-    async fn get_state_heal_paths(&self) -> Result<Option<Vec<(Nibbles, H256)>>, StoreError>;
-
-    /// Set the latest root of the rebuilt state trie and the last downloaded hashes from each segment
-    async fn set_state_trie_rebuild_checkpoint(
-        &self,
-        checkpoint: (H256, [H256; STATE_TRIE_SEGMENTS]),
-    ) -> Result<(), StoreError>;
-
-    /// Get the latest root of the rebuilt state trie and the last downloaded hashes from each segment
-    async fn get_state_trie_rebuild_checkpoint(
-        &self,
-    ) -> Result<Option<(H256, [H256; STATE_TRIE_SEGMENTS])>, StoreError>;
-
-    /// Get the accont hashes and roots of the storage tries awaiting rebuild
-    async fn set_storage_trie_rebuild_pending(
-        &self,
-        pending: Vec<(H256, H256)>,
-    ) -> Result<(), StoreError>;
-
-    /// Get the accont hashes and roots of the storage tries awaiting rebuild
-    async fn get_storage_trie_rebuild_pending(
-        &self,
-    ) -> Result<Option<Vec<(H256, H256)>>, StoreError>;
-
-    /// The `forkchoice_update` and `new_payload` methods require the `latest_valid_hash`
-    /// when processing an invalid payload. To provide this, we must track invalid chains.
-    ///
-    /// We only store the last known valid head upon encountering a bad block,
-    /// rather than tracking every subsequent invalid block.
-    async fn set_latest_valid_ancestor(
-        &self,
-        bad_block: BlockHash,
-        latest_valid: BlockHash,
-    ) -> Result<(), StoreError>;
-
-    /// Returns the latest valid ancestor hash for a given invalid block hash.
-    /// Used to provide `latest_valid_hash` in the Engine API when processing invalid payloads.
-    async fn get_latest_valid_ancestor(
-        &self,
-        block: BlockHash,
-    ) -> Result<Option<BlockHash>, StoreError>;
-
-    /// Obtain block number for a given hash
-    fn get_block_number_sync(
-        &self,
-        block_hash: BlockHash,
-    ) -> Result<Option<BlockNumber>, StoreError>;
-
-    /// Get the canonical block hash for a given block number.
-    fn get_canonical_block_hash_sync(
-        &self,
-        block_number: BlockNumber,
-    ) -> Result<Option<BlockHash>, StoreError>;
+    /// Commits all changes made in this transaction.
+    fn commit(&mut self) -> Result<(), StoreError>;
```

Comment:
This should consume `self`.

Reply:
The tries need this to be a mutable reference to avoid moving the trait object. We can try to change it after merging.